src/docfx/validation/ContentValidator.cs

```diff
@@ -76,7 +87,12 @@ public void ValidateHeadings(FilePath file, List<ContentNode> nodes, bool isIncl
         {
             if (TryGetValidationDocumentType(file, isInclude, out var documentType))
             {
-                var validationContext = new ValidationContext { DocumentType = documentType, FileSourceInfo = new SourceInfo(file) };
+                var validationContext = new ValidationContext
```

Comment:
can we create a method to create ValdiationContext? so we make sure all validation rules have enough validation context

Reply:
Not all validations can have monikers, e.g. manifest files don't have monikers, trying to get monikers on them will cause a warning being thrown.

Reply:
so make the moniker as  optional?